ocs/rook.py

```diff
@@ -204,3 +233,155 @@ def create_cephblockpool(
             raise Exception(err)
 
         return _rc
+
+    def create_cephfs(self, **kwargs):
+        """
+        A function to invoke cephfs creation via CephFS
+
+        Args:
+            **kwargs: Parameter which user want to override against defaults
+                like name, mds count etc. Keys should match as per fs resource
+                yaml.
+
+        Returns:
+            bool: True if fs healthy else False
+        """
+        if self.cephfs:
+            logger.info(f"Cephfs {self.cephfs.name} already exists")
+            return self.cephfs.health_ok()
+        self.cephfs = CephFS(self.namespace, **kwargs)
+        assert self.cephfs.health_ok()
+        return True
+
+
+class CephFS(object):
+    """
+    A class which handles create, delete and modify cephfs
+    No MDS ops here, instead it will be handled by RookCluster class
+    """
+
+    def __init__(self, _api_client, namespace, **kwargs):
+        self._name = kwargs.get('name', default.CEPHFS_NAME)
+        self.namespace = namespace
+        self.CEPHFS =  ocp.OCP(
+            kind='CephFilesystem',
+            namespace=namespace,
+        )
+        self.POD = ocp.OCP(
+            kind='Pod',
+            namespace=namespace,
+        )
+        self.CEPHFS_INITIAL_CONFIG = os.path.join(
+            default.TEMPLATE_DIR, "cephfilesystem.yaml"
+        )
+        self.CEPHFS_CURRENT_CONFIG = "cephfilesystem_tmp.yaml"
+        self.MDS_APP_LABEL = default.MDS_APP_LABEL
+        self.mdss = {}
+        self.active_count = 0
+        #In memory fs configuration(cephfilesystem type)
+        self.conf = None
+        self.data_pool = kwargs.get('data-pool', default.CEPHFS_DATA_POOL)
+
+        resource = generate_yaml_from_jinja2_template_with_data(
+            self.CEPHFS_INITAL_CONFIG,
+            **kwargs,
+        )
+
+        with open(self.CEPHFS_CURRENT_CONFIG, 'w') as conf:
+            yaml.dump(resource, conf, default_flow_style=False)
+
+        logger.info(f"Creating ceph FileSystem {self._name}")
```

Comment:
please move this implementation to create() method.
In addition, I don't think the creation should be part of the init().
If a user wants, this can be done with CephFS(**kwargs).create()